### PR TITLE
fix: make typescript test example spec compliant

### PR DESCRIPTION
**Pull request by SimenB**
### Proposed changes

See https://github.com/pulumi/examples/pull/608#discussion_r400767835

### CODE REVIEW

**Comment by stack72** [comment]
@komalali thoughts? I know you are making some improvements in some areas for this

**Comment by komalali** [comment]
I haven't touched anything specific to this particular case, but the reasoning looks sound to me following the conversation at https://github.com/pulumi/examples/pull/608#discussion_r400767835

@mikhailshilkov was involved in the original convo so tagging him here, but looks good to me and thanks for the improvement @SimenB !

File: content/docs/guides/testing/unit.md [hunk 1]
```diff
@@ -330,18 +330,22 @@ pulumi.runtime.setMocks({
     // ... mocks as shown above
 });
 
-// It's important to import the program _after_ the mocks are defined.
-import * as infra from "./index";
-
 describe("Infrastructure", function() {
-    const server = infra.server;
+    let infra: typeof import('../index')
```
Comment (mikhailshilkov) [inline content/docs/guides/testing/unit.md:334]:
```suggestion
    let infra: typeof import('../index');
```

File: content/docs/guides/testing/unit.md [hunk 2]
```diff
@@ -330,18 +330,22 @@ pulumi.runtime.setMocks({
...
+
+    before(async function() {
+        // It's important to import the program _after_ the mocks are defined.
+        infra = await import('../index')
```
Comment (mikhailshilkov) [inline content/docs/guides/testing/unit.md:338]:
```suggestion
        infra = await import('../index');
```

File: content/docs/guides/testing/unit.md [hunk 3]
```diff
@@ -330,18 +330,22 @@ pulumi.runtime.setMocks({
...
-    const server = infra.server;
+    let infra: typeof import('../index');
+
+    before(async function() {
+        // It's important to import the program _after_ the mocks are defined.
+        infra = await import('../index');
+    })
+
     describe("#server", function() {
+        const server = infra.server;
```
Comment (SimenB) [inline content/docs/guides/testing/unit.md:342]: looking back, this is broken. We cannot destructure inside `describe` as that runs before any hooks (like `before`) and tests. It needs to be accessed either after the `await` or inside a test

Comment (SimenB) [reply content/docs/guides/testing/unit.md:342]: (at least that's the way Jest and Jasmine works, I don't think I've every used Mocha. But I assume it also works that way)

**Pull request merged**